File: Card_Game_Engine/Models/Classes/Trigger.cs
namespace Card_Game_Engine.Models.Classes;

public class Trigger
{
    public int Id { get; set; }
    public List<Parameter> Parameters { get; set; } = new List<Parameter>();

    public override string ToString()
    {
        var parameters = string.Join(", ", Parameters.Select(p => p.ToString()));
        return $"Trigger Id: {Id}, Parameters: [{parameters}]";
    }
}

File: Card_Game_Engine/ClientApp/src/app/_reusable-components/trigger/namespace/classes/trigger.ts
import {Parameter} from "../../../parameter/namespace/classes/parameter";

export class Trigger {
  id : number;
  display: string;
  parameters: Parameter[];

  constructor(id: number, display: string, parameters: Parameter[] = []) {
    this.id = id;
    this.display = display;
    this.parameters = parameters;
  }
}
